### PR TITLE
Newspaper bap fix

### LINKED ISSUE #24413: Pets can be bapped with a "rolled up" newspaper that wasn't rolled up
### Exploit Reports

- [X] I confirm this issue is not an exploit. (Required)

### BYOND Version

515.1623

### Issue Description

Pets can be bapped with a "rolled up" newspaper that wasn't rolled up

### What did you expect to happen?

If the newspaper isn't rolled up, no bapping.

### What happened instead?

Bapping was had.

### Why is this bad/What are the consequences?

It's inconsistent with the displayed text.

### Steps to reproduce the issue.

1. Spawn as HoP.
2. Print out a newspaper at a newscaster.
3. Take the newspaper.
4. Bap Ian with the newspaper.
5. What do you mean it was rolled up?

### When did the problem start happening?

Unknown

### Extra information

Code is here: https://github.com/ParadiseSS13/Paradise/blob/7ae3bf6747a61768a8b2abf7f01ee6aa8a5e90c8/code/modules/mob/living/simple_animal/friendly/pet.dm#L9

### Relevant log output/runtime error

_No response_

### PR DESCRIPTION
**Pull request by EmeraldCandy**
## What Does This PR Do
fixes #24413

Properly gets pet bap message when hit on the nose with a rolled up newspaper

## Why It's Good For The Game
bugfixes good

## Testing
rolled up newspaper
bapped ian
bapped renault
bapped some skrells
did previous unrolled

## Changelog
:cl:
fix: Fixed bapping pets with rolled newspaper
/:cl:

### CODE REVIEW

**Review by Spaghetti-bit — APPROVED** [review]

File: code/modules/mob/living/simple_animal/friendly/pet.dm [hunk 1]
```diff
@@ -9,11 +9,13 @@
...
 	if(istype(O, /obj/item/newspaper))
-		if(!stat)
+		var/obj/item/newspaper/paper = O
+		if(!stat && paper.rolled)
 			user.visible_message("<span class='notice'>[user] baps [name] on the nose with the rolled up [O].</span>")
 			spawn(0)
 				for(var/i in list(1,2,4,8,4,2,1,2))
 					setDir(i)
 					sleep(1)
+		else return ..()
```
Comment (Sirryan2002) [inline code/modules/mob/living/simple_animal/friendly/pet.dm:19]:
```suggestion
		else
			return ..()
```
even tho it looks cleaner, should still follow standard formatting rules

**Review by Sirryan2002 — APPROVED** [review]

**Review by DGamerL — CHANGES_REQUESTED** [review]
Yeah, please turn both of these `if`-statements into early returns

**Comment by EmeraldCandy** [comment]
> Yeah, please turn both of these `if`-statements into early returns

like this yes?

File: code/modules/mob/living/simple_animal/friendly/pet.dm [hunk 2]
```diff
@@ -9,11 +9,14 @@
...
 			user.visible_message("<span class='notice'>[user] baps [name] on the nose with the rolled up [O].</span>")
 			spawn(0)
 				for(var/i in list(1,2,4,8,4,2,1,2))
 					setDir(i)
 					sleep(1)
-	else
+			return ..()
 		return ..()
+
+
```
[the remark this answers is not in the document]
Comment (EmeraldCandy) [reply code/modules/mob/living/simple_animal/friendly/pet.dm:22]: oh, i see thank you.

**Review by DGamerL — APPROVED** [review]
This proc isquite atrocious

File: code/modules/mob/living/simple_animal/friendly/pet.dm [hunk 3]
```diff
@@ -8,12 +8,15 @@
...
-				for(var/i in list(1,2,4,8,4,2,1,2))
-					setDir(i)
-					sleep(1)
-	else
+	if(!istype(O, /obj/item/newspaper))
 		return ..()
+	if(stat != CONSCIOUS || !paper.rolled)
+		return
+	var/obj/item/newspaper/paper = O
+	user.visible_message("<span class='notice'>[user] baps [name] on the nose with the rolled up [O].</span>")
```
Comment (DGamerL) [inline code/modules/mob/living/simple_animal/friendly/pet.dm:16]:
```suggestion
	var/obj/item/newspaper/paper = O
	if(stat != CONSCIOUS || !paper.rolled)
		return
	user.visible_message("<span class='notice'>[user] baps [name] on the nose with the rolled up [O].</span>")
```
I'm stupid

File: code/modules/mob/living/simple_animal/friendly/pet.dm [hunk 4]
```diff
@@ -8,12 +8,15 @@
...
+	if(!istype(O, /obj/item/newspaper))
 		return ..()
+	var/obj/item/newspaper/paper = O
+	if(stat != CONSCIOUS || !paper.rolled)
+		return
+	user.visible_message("<span class='notice'>[user] baps [name] on the nose with the rolled up [O].</span>")
+	spawn(0)
+		for(var/i in list(1,2,4,8,4,2,1,2))
+			setDir(i)
+			sleep(1)
```
Comment (hal9000PR) [inline code/modules/mob/living/simple_animal/friendly/pet.dm:20]: please use the `spin` proc

**Review by hal9000PR — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/simple_animal/friendly/pet.dm (hunk 4 above)
Comment (DGamerL) [reply code/modules/mob/living/simple_animal/friendly/pet.dm:20]: Out of scope, this is a fix, not a full refactor

**Review by hal9000PR — APPROVED** [review]

**Review by Burzah — APPROVED** [review]

**Pull request merged**